### PR TITLE
Fix CI/CD and build the compatible binary connecting wallet as plug-in to thyra

### PR DESCRIPTION
**Pull request by B-Naoufal**
1- Write a script called build.sh that will create the thyra-plugin-wallet Binary, taking in two arguments: the port number and the path to the plugin.  
2- Fix The CD calling the build.sh, and upload artifacts for multiple Operating Systems.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -ex
+# Set the GOPATH environment variable
+PLUGIN=wallet-plugin
+
+
+# Build the binary
+mkdir -p "build/$PLUGIN/"
+env CGO_ENABLED="1" go build -o build/$PLUGIN/thyra-plugin-wallet  cmd/massa-wallet/main.go
```
Comment (peterjah) [inline build.sh:9]: what is "env". i think you can remove it

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,71 +1,86 @@
...
+    # The lint job uses a configuration file in the .github/workflows directory.
     uses: ./.github/workflows/lint.yml
 
   tests:
+    # The tests job uses a configuration file in the .github/workflows directory.
     uses: ./.github/workflows/tests.yml
 
   build:
-    if: ${{ github.ref == 'refs/heads/main' }}
+    # The build job depends on the lint and tests jobs to run successfully before it can start.
```
Comment (peterjah) [inline .github/workflows/build.yml:28]: please add this:
    if: ${{ (github.ref == 'refs/heads/main') || startsWith(github.ref, 'refs/tags/v') }}

We want to build binaries only on main branch and releases

**Review by peterjah — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (B-Naoufal) [reply .github/workflows/build.yml:28]: ok but we may have some broken CI in main after merge :? how to test that the build is working well in all targets before merge ? @peterjah

Comment (B-Naoufal) [reply .github/workflows/build.yml:28]: yes you're right

**Review by peterjah — APPROVED** [review]

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -ex
```
Comment (gregLibert) [inline build.sh:2]:
```suggestion
#!/bin/bash -ex
```

File: cmd/massa-wallet/main.go [hunk 4]
```diff
@@ -14,14 +15,19 @@ import (
 )
 
 func main() {
-	myApp := app.New()
+	var port int
+	var path string
+	flag.IntVar(&port, "port", 8080, "the port to listen on")
+	flag.StringVar(&path, "path", "", "the path to listen on")
```
Comment (gregLibert) [inline cmd/massa-wallet/main.go:21]: Why ?

**Review by gregLibert — CHANGES_REQUESTED** [review]
:mag: nitpicking and unused variable

**Review by gregLibert — APPROVED** [review]
:clap:

**Comment by SlnPons** [comment]
@B-Naoufal same question as for #11 not sure how to test it correctly.

**Pull request merged**